### ISSUE TITLE
Static query builder methods and local scopes not resolved on Model classes

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#498]
## Description

Eloquent models forward static method calls to the query builder via `__callStatic`. Common patterns like `Model::whereIn(...)`, `Model::where(...)`, `Model::orderBy(...)` are reported as `UndefinedMagicMethod` because the plugin doesn't resolve these delegated calls.

Similarly, local scopes defined as `scopeX()` methods are callable as `Model::x()` but the plugin doesn't resolve them.

## Impact

**15 false positives** in [monicahq/monica](https://github.com/monicahq/monica). While the count is modest in this codebase, these patterns are extremely common across Laravel applications:
- `Model::whereIn()` — 8 occurrences across 6 models
- `Model::where()` — common but already handled at errorLevel 3
- `Model::active()` (scope) — 1 occurrence
- `Model::increment()` — 1 occurrence

## Reproduction

Found while auditing [monicahq/monica](https://github.com/monicahq/monica) at Psalm errorLevel 1.

### Static query builder method

**File:** `app/Domains/Settings/CancelAccount/Services/CancelAccount.php:51`

```php
$vaultIds = $account->vaults()->select('id')->get()->toArray();

// UndefinedMagicMethod: Magic method App\Models\File::wherein does not exist
File::whereIn('vault_id', $vaultIds)->chunk(100, function ($files) {
    $files->each(function ($file) {
        $file->delete();
    });
});
```

### Local scope

**Model** (`app/Models/AddressBookSubscription.php:175`):
```php
public function scopeActive(Builder $query): Builder
{
    return $query->where('active', true);
}
```

**Usage** (`app/Domains/Contact/DavClient/Jobs/UpdateAddressBooks.php:23`):
```php
// UndefinedMagicMethod: Magic method App\Models\AddressBookSubscription::active does not exist
AddressBookSubscription::active()->get();
```

## Expected Behavior

1. `File::whereIn(...)` should resolve to `Builder::whereIn(...)` and return `Builder<File>`
2. `AddressBookSubscription::active()` should resolve to the `scopeActive()` method on the model

## Proposed Fix

Two separate mechanisms:

1. **Static query builder forwarding**: When a static method is called on an Eloquent Model subclass and the method exists on `Illuminate\Database\Eloquent\Builder`, resolve it as a call to `Builder::methodName()` with the appropriate generic type.

2. **Local scopes**: Scan model classes for `scope*` methods and register the corresponding magic methods (without the `scope` prefix, lowercased first letter). The return type should be `Builder<ModelClass>`.
